packages/rum-vue/test/e2e/app.e2e-spec.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2017-present, Elasticsearch BV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+const {
+  waitForApmServerCalls,
+  getBrowserInfo
+} = require('../../../../dev-utils/webdriver')
+
+describe('Vue router integration', function() {
+  beforeAll(() => browser.url('/test/e2e/'))
+
+  it('should run vue app and capture route-change events', function() {
+    browser.waitUntil(
+      () => {
+        /**
+         * route to /fetch
+         */
+        $('#fetch').click()
+        const fetchResult = $('#content')
+        return fetchResult.getText().indexOf('loaded data.json') !== -1
+      },
+      5000,
+      'expected data.json to be loaded'
+    )
+
+    const serverCalls = waitForApmServerCalls(0, 2)
+    expect(serverCalls.sendTransactions.length).toBe(2)
+
+    const pageLoadTransaction = serverCalls.sendTransactions[0].args[0][0]
+    expect(pageLoadTransaction.type).toBe('page-load')
+    expect(pageLoadTransaction.name).toBe('/')
+    expect(pageLoadTransaction.spans.length).toBeGreaterThan(1)
+
+    const routeTransaction = serverCalls.sendTransactions[1].args[0][0]
+    expect(routeTransaction.name).toBe('/fetch')
+    expect(routeTransaction.type).toBe('route-change')
+
+    const spanTypes = ['app', 'external']
+    const foundSpans = routeTransaction.spans.filter(
+      span => spanTypes.indexOf(span.type) > -1
+    )
+    /**
+     * `app` span type will not be captured in safari 9 since
+     * User Timing API is not supported.
+     */
+    const { name } = getBrowserInfo()
+    if (name.indexOf('safari') >= 0) {
+      expect(foundSpans.length).toEqual(1)
+    } else {
+      expect(foundSpans.length).toEqual(2)
+    }
+  })
```

Comment:
Can we use something else instead of user timing API (like creating a span by calling `$apm.startSpan()`)?

This adds technical debt that we need to clean up later on.